core/nonce.go

```diff
@@ -94,10 +100,15 @@ func (ns NonceService) decrypt(nonce string) (int64, error) {
 
 // Nonce provides a new Nonce.
 func (ns *NonceService) Nonce() (string, error) {
+	ns.mu.Lock()
 	ns.latest++
-	return ns.encrypt(ns.latest)
+	latest := ns.latest
+	ns.mu.Unlock()
+	return ns.encrypt(latest)
 }
 
+// minUsed returns the lowest key in the used map. Requires that a lock be held
```

Comment:
This "Requires that a lock be held by the caller" is a bit confusing since you do the locking in here.